### PR TITLE
feat(resize/volume): add openapi spec and REST service interface

### PR DESCRIPTION
**Pull request by dsharma-dc**

### CODE REVIEW

**Comment by dsharma-dc** [comment]
bors try

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 1]
```diff
@@ -1527,6 +1527,38 @@ paths:
           $ref: '#/components/responses/ServerError'
       security:
         - JWT: []
+  '/volumes/{volume_id}/resize':
+    put:
+      tags:
+        - Volumes
+      operationId: put_volume_resize
```
Comment (tiagolobocastro) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml]:
```suggestion
  '/volumes/{volume_id}/size':
    put:
      tags:
        - Volumes
      operationId: put_volume_size
```

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 2]
```diff
@@ -2369,6 +2401,28 @@ components:
...
+        capacity_limit: 204857610
+      description: Resize Volume Body
+      type: object
+      properties:
+        size:
+          description: Required new size of the volume in bytes
+          type: integer
+          format: uint64
+          minimum: 0
+        capacity_limit:
```
Comment (tiagolobocastro) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml]: We're not using this today, let's omit it and add it when we needed?

Comment (dsharma-dc) [reply control-plane/rest/openapi-specs/v0_api_spec.yaml]: fair enough. Removed for now. 
cc: @chriswldenyer

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 3]
```diff
@@ -2369,6 +2401,19 @@ components:
...
         - snapshot
         - volume
+    ResizeVolumeBody:
+      example:
+        size: 104857610
+      description: Resize Volume Body
+      type: object
+      properties:
+        size:
+          description: Required new size of the volume in bytes
```
Comment (Abhinandan-Purkait) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml]: Nit: `New required size of the volume in bytes` ?

File: control-plane/rest/service/src/v0/volumes.rs [hunk 4]
```diff
@@ -196,6 +196,24 @@ impl apis::actix_server::Volumes for RestApi {
...
+    async fn put_volume_size(
+        Path(volume_id): Path<Uuid>,
+        Body(resize_volume_body): Body<models::ResizeVolumeBody>,
+    ) -> Result<models::Volume, RestError<RestJsonError>> {
+        let volume = client()
+            .resize(
+                &ResizeVolume {
+                    uuid: volume_id.into(),
+                    requested_size: resize_volume_body.size as u64,
+                    capacity_limit: None,
```
Comment (Abhinandan-Purkait) [inline control-plane/rest/service/src/v0/volumes.rs:208]: Is this field not exposed via rest?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/rest/service/src/v0/volumes.rs (hunk 4 above)
Comment (Abhinandan-Purkait) [reply control-plane/rest/service/src/v0/volumes.rs:208]: Please ignore, seems like it's covered in the previous review comments.

Comment (dsharma-dc) [reply control-plane/rest/service/src/v0/volumes.rs:208]: Not for now, as per this comment:
https://github.com/openebs/mayastor-control-plane/pull/700#discussion_r1431311295

**Review by niladrih — APPROVED** [review]

**Comment by dsharma-dc** [comment]
bors merge

**Pull request merged**